File: app/src/main/java/com/justnik/mirea_viewbinding_lesson/ExampleFragment.java
package com.justnik.mirea_viewbinding_lesson;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.justnik.mirea_viewbinding_lesson.databinding.FragmentExampleBinding;


public class ExampleFragment extends Fragment {

    private FragmentExampleBinding binding;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentExampleBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        binding.bExample.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String text = binding.etExample.getText().toString();
                if (text.isEmpty()){
                    binding.etExample.setError("Пустое поле");
                }
                else{
                    binding.tvExample.setText(text);
                }
            }
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }

    public static ExampleFragment newInstance() {
        ExampleFragment fragment = new ExampleFragment();

        //Здесь вы можете создать Bundle и добавить его во фрагмент

        return fragment;
    }
}

File: app/src/main/java/com/justnik/mirea_viewbinding_lesson/ExampleFragment.kt
package com.justnik.mirea_viewbinding_lesson

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import by.kirich1409.viewbindingdelegate.viewBinding
import com.justnik.mirea_viewbinding_lesson.databinding.FragmentExampleBinding

class ExampleFragment : Fragment(R.layout.fragment_example) {
    private val binding: FragmentExampleBinding by viewBinding()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        setClickListener()
    }

    private fun setClickListener() {
        with(binding) {
            bExample.setOnClickListener {
                val text = etExample.text.toString()
                if(text.isEmpty()){
                    etExample.error = "Пустое поле"
                }
                else{
                    tvExample.text = text
                }
            }
        }
    }

    companion object {
        @JvmStatic
        fun newInstance(): ExampleFragment {
            val fragment = ExampleFragment()

            //Здесь вы можете создать Bundle и добавить его во фрагмент

            return fragment
        }
    }
}